LeaderboardBackend/Services/Impl/EmailSender.cs

```diff
@@ -0,0 +1,62 @@
+using MailKit.Net.Smtp;
+using Microsoft.Extensions.Options;
+using MimeKit;
+
+namespace LeaderboardBackend.Services;
+
+public class EmailSender : IEmailSender
+{
+    private readonly EmailSenderConfig _config;
+    private readonly ILogger<EmailSender> _logger;
+    private readonly ISmtpClient _smtpClient;
+    private readonly MailboxAddress _sender;
+
+    public EmailSender(IOptions<EmailSenderConfig> config, ILogger<EmailSender> logger, ISmtpClient smtpClient)
+    {
+        _config = config.Value;
+        _logger = logger;
+        _smtpClient = smtpClient;
+        _sender = new MailboxAddress(name: _config.SenderName, address: _config.SenderAddress);
+    }
+
+    public async Task SendEmailAsync(string recipientAddress, string subject, string htmlMessage)
+    {
+        if (_config.Smtp is null)
+        {
+            _logger.LogWarning("Can't send email (subject: \"{EmailSubject}\"), SMTP configuration is missing",
+                subject);
+            return;
+        }
+
+        MimeMessage message = new();
+        message.From.Add(_sender);
+        message.To.Add(new MailboxAddress(name: null, address: recipientAddress));
+        message.Subject = subject;
+
+        BodyBuilder bodyBuilder = new() { HtmlBody = htmlMessage };
+        message.Body = bodyBuilder.ToMessageBody();
+
+        try
+        {
+            await SendEmail(message);
+            _logger.LogInformation("Sent email (subject \"{EmailSubject}\")", subject);
```

Comment:
I'm not sure this is worth logging because the subjects of the emails we send are going to be very generic, along the lines of "Recover Your Account" or "Confirm Your Account." It might be better to log from whatever method calls the email sending method so it can give a helpful log message like "sent recoery email to account ace5bf6885baf6" instead.